demo/util/util.mjs

```diff
@@ -0,0 +1,14 @@
+/**
+ * Replaces |swapOut| with |swapIn| in the DOM. Reparent all children
+ * and copy all attributes.
+*/
+export function swap(swapOut, swapIn) {
+  while (swapOut.childNodes.length > 0) {
+    swapIn.appendChild(swapOut.firstChild);
+  }
+  for (const a of swapOut.getAttributeNames()) {
+    swapIn.setAttribute(a, swapOut.getAttribute(a));
+  }
+  swapOut.parentElement.insertBefore(swapIn, swapOut);
```

Comment:
I think these two lines can be changed to `swapOut.replaceWith(swapIn)`

Reply:
done